soroban-env-host/src/host/conversion.rs

```diff
@@ -490,7 +501,13 @@ impl Host {
             }
             ScVal::Bytes(b) => Ok(self.add_host_object(b.metered_clone(self)?)?.into()),
             ScVal::String(s) => Ok(self.add_host_object(s.metered_clone(self)?)?.into()),
-            ScVal::Symbol(s) => Ok(self.add_host_object(s.metered_clone(self)?)?.into()),
+            ScVal::Symbol(s) => Ok(self
+                .add_host_object(ScSymbol::try_from_bytes(
```

Comment:
Could you add a comment here explaining why this is necessary so future refactors / maintainers don't remove it?

Reply:
Done.